subsys/net/ip/net_tls.c

```diff
@@ -0,0 +1,735 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_NET_TLS_DEBUG)
+#define SYS_LOG_DOMAIN "net/tls"
+#define NET_LOG_ENABLED 1
+#endif
+
+#include <stdio.h>
+#include <sys/param.h>
+
+#include "net_private.h"
+#include "net_tls_internal.h"
+#include "tcp_internal.h"
+
+#define TIMEOUT_TLS_RX_MS       100
+#define TIMEOUT_TLS_TX_MS       100
+
+#if defined(CONFIG_NET_PRECONFIGURE_TLS_CREDENTIALS)
+#include CONFIG_NET_TLS_CREDENTIALS_FILE
+#endif
+
+static mbedtls_ctr_drbg_context tls_ctr_drbg;
+
+struct net_tls_credential {
+	enum net_tls_credential_type type;
+	sec_tag_t tag;
+	const void *buf;
+	size_t len;
+};
+
+/* Global poll of credentials shared among TLS contexts. */
+static struct net_tls_credential credentials[CONFIG_NET_MAX_CREDENTIALS_NUMBER];
+
+/* A poll of TLS contexts. */
+static struct net_tls tls_context[CONFIG_NET_TLS_MAX_CONTEXT];
+
+#if defined(MBEDTLS_DEBUG_C) && defined(CONFIG_NET_TLS_DEBUG)
+static void my_debug(void *ctx, int level,
```

Comment:
Maybe not my_debug() ;-).

Reply:
It's actually a copy-paste of existing Zephyr code (net_app.c). If it has issues, it should be fixed though.

Reply:
> It's actually a copy-paste of existing Zephyr code (net_app.c).

I also had such copy-pastes in Zstream code. I had to fix them up before removing "[WIP]" from the PR title though ;-).